### ISSUE TITLE
selectre chokes on short rows

### ISSUE THREAD

**Issue by alimanfoo** [alimanfoo/petl#90]
```
In [27]: rowcount(t4)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
/media/truecrypt2/Documents/2012/05/10/<ipython-input-27-113e1ee7895e> in <module>()
----> 1 rowcount(t4)

/usr/local/lib/python2.7/dist-packages/petl/__init__.pyc in wrapper(*args, **kwargs)
     79 def wrap(f):
     80     def wrapper(*args, **kwargs):
---> 81         _innerresult = f(*args, **kwargs)
     82         if isinstance(_innerresult, RowContainer):
     83             return RowContainerWrapper(_innerresult)

/usr/local/lib/python2.7/dist-packages/petl/util.pyc in rowcount(table)
    174 
    175     n = 0
--> 176     for row in data(table):
    177         n += 1
    178     return n

/usr/local/lib/python2.7/dist-packages/petl/__init__.pyc in _iterwithcache(self)
     49         if debug: print repr(self._inner) + ' :: cache exhausted, serving from inner iterator'
     50         it = iter(self._inner)
---> 51         for row in islice(it, len(self._cache), None):
     52             # maybe there's more room in the cache?

     53             if len(self._cache) < cachesize:

/usr/local/lib/python2.7/dist-packages/petl/transform.pyc in iterfieldselect(source, field, where, complement)
   2969     getv = itemgetter(*indices)
   2970     for row in it:
-> 2971         v = getv(row)
   2972         if where(v) != complement: # XOR
   2973             yield tuple(row)

IndexError: tuple index out of range
```

**Comment by alimanfoo** [comment]
Also review selectre versus search, redundant?

**Comment by alimanfoo** [comment]
selectre() is redundant, duplicates functionality available via search(), will be removed in v1.0
